jabgui/src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -562,4 +607,167 @@ public void nextPreviewStyle() {
+            findFirstFocusableNode(searchRoot).ifPresent(Node::requestFocus);
+        }
+    }
+
+    /// Recursively searches for a TextInputControl (TextField or TextArea) with the given ID.
+    private Optional<TextInputControl> findTextInputById(Parent parent, String id) {
+        for (Node child : parent.getChildrenUnmodifiable()) {
+            if (child instanceof TextInputControl textInput && id.equalsIgnoreCase(textInput.getId())) {
+                return Optional.of(textInput);
+            } else if (child instanceof Parent childParent) {
+                Optional<TextInputControl> found = findTextInputById(childParent, id);
+                if (found.isPresent()) {
+                    return found;
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<TextInputControl> findAnyTextInput(Parent parent) {
+        for (Node child : parent.getChildrenUnmodifiable()) {
+            if (child instanceof TextInputControl textInput) {
+                return Optional.of(textInput);
+            } else if (child instanceof Parent childParent) {
+                Optional<TextInputControl> found = findAnyTextInput(childParent);
+                if (found.isPresent()) {
+                    return found;
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    // Returns the first focusable, visible, managed, and enabled node in depth-first order
+    private Optional<Node> findFirstFocusableNode(Parent parent) {
+        for (Node child : parent.getChildrenUnmodifiable()) {
+            if (isNodeFocusable(child)) {
+                return Optional.of(child);
+            } else if (child instanceof Parent childParent) {
+                Optional<Node> found = findFirstFocusableNode(childParent);
+                if (found.isPresent()) {
+                    return found;
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    // Returns the last focusable, visible, managed, and enabled node in depth-first order
+    private Optional<Node> findLastFocusableNode(Parent parent) {
+        Optional<Node> last = Optional.empty();
+        for (Node child : parent.getChildrenUnmodifiable()) {
+            if (child instanceof Parent childParent) {
+                Optional<Node> sub = findLastFocusableNode(childParent);
+                if (sub.isPresent()) {
+                    last = sub;
+                }
+            }
+            if (isNodeFocusable(child)) {
+                last = Optional.of(child);
+            }
+        }
+        return last;
+    }
+
+    private boolean isNodeFocusable(Node node) {
+        return node.isFocusTraversable() && node.isVisible() && !node.isDisabled() && node.isManaged();
+    }
+
+    // Tries to locate the editor grid (with style class "editorPane") inside the tab content to avoid
+    // including preview or other sibling panels when determining focus order boundaries.
```

Comment:
```suggestion
    /// Tries to locate the editor grid (with style class "editorPane") inside the tab content to avoid
    /// including preview or other sibling panels when determining focus order boundaries.
```